src/main/java/MarkTask.java

```diff
@@ -0,0 +1,48 @@
+import java.nio.InvalidMarkException;
+import java.util.ArrayList;
+
+public class MarkTask {
+    public static int findIndexToMark(String[] userInputWords, ArrayList<Task> tasks) throws
+            DukeExceptions.InvalidItemException, ArrayIndexOutOfBoundsException {
+        try {
+            int testIndex = Integer.parseInt(userInputWords[1]);
+        } catch(Exception e) {
+            DukeExceptions.throwInvalidItemException();
+        }
+        int indexFound = Integer.parseInt(userInputWords[1]);
+        if (indexFound > tasks.size() || indexFound < 1) {
+            throw new ArrayIndexOutOfBoundsException();
+        }
+        return indexFound;
+    }
+
+    public static void mark(String[] userInputWords, ArrayList<Task> tasks) {
+        int indexToMark = -1;
+        String isMarked = userInputWords[0];
+        try {
+            indexToMark = findIndexToMark(userInputWords, tasks);
+        } catch(ArrayIndexOutOfBoundsException e) {
```

Comment:
```suggestion
        } catch (ArrayIndexOutOfBoundsException e) {
```
do take note of other missing spaces in your code